SConstruct

```diff
@@ -187,26 +224,35 @@ if env["platform"] == "linux" or env["platform"] == "freebsd":
     elif env["target"] == "release":
         env.Append(CCFLAGS=["-O3"])
 
-    if env["bits"] == "64":
-        env.Append(CCFLAGS=["-m64"])
-        env.Append(LINKFLAGS=["-m64"])
-    elif env["bits"] == "32":
-        env.Append(CCFLAGS=["-m32"])
-        env.Append(LINKFLAGS=["-m32"])
+    if env["arch"] == "x86_64":
+        # -m64 and -m32 are x86-specific already, but it doesn't hurt to
+        # be clear and also specify -march=x86_64. Similar with 32-bit.
+        env.Append(CCFLAGS=["-m64", "-march=x86-64"])
+        env.Append(LINKFLAGS=["-m64", "-march=x86-64"])
+    elif env["arch"] == "x86_32":
+        env.Append(CCFLAGS=["-m32", "-march=i686"])
+        env.Append(LINKFLAGS=["-m32", "-march=i686"])
```

Comment:
Is there any need to keep `-mBITS` at all, `-march` is probably enough.

Reply:
I just tested, it is needed. With only `-march=i686`, GCC says `cc1plus: error: CPU you selected does not support x86-64 instruction set`. It's a bit weird, the intuitive thing would be that only `-march=` should be needed since that's all we need for ARM and RISC-V, but I guess it's a compiler quirk we have to deal with.